docs/user/authoring-input-modes.rst

```diff
@@ -0,0 +1,69 @@
+* **Resolution** - the data resolution. Higher gives more detail but takes more memory.
+* **Brush Radius** - controls the radius of the brush cursor used for painting. Visualised by the cursor in the Scene View.
+* **Brush Strength** - scales intensity of the brush interaction.
+
+Some planning is advisable before committing significant time to creating the input. The data is not currently resampled when the size/position/resolution is changed, so for example the data needs to be re-authored if the resolution is changed.
+
+As a final note, painted data will be saved into the scene with the component. Consider minimising data resolution and painting multiple small areas rather than one huge area whenever it makes sense. The storage required by saving this data may be evident by a sudden increase in the size of the scene file. While most painted input is likely to compress well into a small package size, it may still be worth creating prefabs from painted input GameObjects so that the data is stored outside the main level file.
+
+
+# Spline Mode
+
+Selecting Spline from the Mode dropdown will enable Spline Mode Settings UI in the Inspector. This mode requires a Spline component to be present with at least 2 spline points added. Help boxes in the Inspector serve to guide/automate this setup.
+
+Once a Spline is created, this is used to drive the input. A common use of splines is to set the water level to follow a riverbed using the RegisterHeightInput component. A spline may also be used to add waves or flow velocity, if this gives the required level of fidelity (if not then painting may be a good option, described in the previous section). Another typical use case of splines is to add waves aligned to shorelines.
+
+Relevant data components will automatically be added to spline points. For example if the spline is used with a RegisterFlowInput component, the Spline Point Data Flow component will be added to spline points which can then be used to configure the flow speed.
+
+
+# Custom Geometry and Shader
+
+This is the most advanced type of input and allows rendering any geometry/shader into the water system data. One could draw foam directly into the foam data, or inject a flow map baked from an offline sim.
+
+The geometry can come from a standard MeshRenderer/MeshFilter combination, or it can come from any *Renderer* component such as a particle system. 
+This geometry will be rendered from a orthographic top down perspective to "print" the data onto the water.
+
+The *Particle Renderer* example in the Examples scene shows a particle system being projected onto the water surface.
+
+
+# Primitive
+
+Use a primitive shape for the input such as a mathematical cube or sphere.
+
+At time of writing used for surface clipping - the volume defined by the primitive shape will be cut from the water surface.
```

Comment:
"At time of writing used for surface clipping" Little awkward/confusing.

Reply:
agreed fixed